prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1489,7 +1489,7 @@ removeTrailingWhitespace = go (RecordedWhitespace [] 0)
     -- release only the necessary ones.
     go (RecordedWhitespace withheldLines withheldSpaces) = \sds -> case sds of
         SFail -> SFail
-        SEmpty -> SEmpty
+        SEmpty -> foldr (\_i sds' -> SLine 0 sds') SEmpty withheldLines
```

Comment:
Question here is: should there only be a single newline remaining (to comply with Unix best practices), rather than all of them?

Reply:
FWIW I decided to keep only a single trailing newline.

Reply:
This seems to work for `dhall-haskell`, and I kind of understand the reasoning but it feels a tad arbitrary to me:

While it's easy to introduce trailing spaces (` `) inadvertedly by using the `prettyprinter` combinators, trailing newlines are more likely to be added on purpose.

In any case, I think the documentation should be updated:

https://github.com/quchen/prettyprinter/blob/0fd83e7f8ff50bc1e22f665bb71819b3e1a1c3af/prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs#L1451

To me this sounds as if the function is only concerned about single space characters (` `).

Reply:
Hm, you’re right actually, the function should not be concerned with newlines. I’ll revert it back to your version.

Reply:
Thanks! :)